/**:
  ros__parameters:
    use_sim_time: True
    map_frame_id: "map"
    odom_frame_id: "odom"
    plane_extraction_frame_id: body
    plane_visualization_frame_id: body_elevated
    use_map2map_transform: false   # True if loading a previous posegraph for multi-session
    # use_tts: true

    # Optimization params
    g2o_solver_type: "lm_var_cholmod" # gn_var, gn_fix6_3, gn_var_cholmod, lm_var, lm_fix6_3, lm_var_cholmod
    # g2o_solver_type: "gn_var_cholmod" # gn_var, gn_fix6_3, gn_var_cholmod, lm_var, lm_fix6_3, lm_var_cholmod
    g2o_solver_num_iterations: 512

    # Constraint switches
    enable_gps: false
    enable_imu_acceleration: false
    enable_imu_orientation: false

    # Keyframe registration
    max_keyframes_per_update: 50
    keyframe_delta_trans: 2.0
    keyframe_delta_angle: 2.0

    # Fix first node for optimization stability
    fix_first_node: true
    fix_first_node_stddev: 1 1 1 1 1 1
    fix_first_node_adaptive: true

    # Loop closure
    distance_thresh: 1.0
    accum_distance_thresh: 1.0
    min_edge_interval: 1.0
    fitness_score_thresh: 0.5
    roompart_length: 5.0

    # room vertex
    cluster_resolution: 0.5


    # Scan matching
    registration_method: "FAST_GICP"
    reg_num_threads: 8
    reg_transformation_epsilon: 0.01
    reg_maximum_iterations: 64
    reg_max_correspondence_distance: 2.5
    reg_max_optimizer_iterations: 20
    reg_use_reciprocal_correspondences: false
    reg_correspondence_randomness: 20
    reg_resolution: 1.0
    reg_nn_search_method: "DIRECT7"
        
    # Edge parameters
        
    ## GPS
    gps_edge_robust_kernel: "NONE"
    gps_edge_robust_kernel_size: 1.0
    gps_edge_stddev_xy: 20.0
    gps_edge_stddev_z: 5.0

    ## IMU orientation
    imu_orientation_edge_robust_kernel: "NONE"
    imu_orientation_edge_stddev: 1.0

    ## IMU acceleration
    imu_acceleration_edge_robust_kernel: "NONE"
    imu_acceleration_edge_stddev: 1.0

    # Ground plane
    floor_edge_robust_kernel: "NONE"
    floor_edge_stddev: 10.0

    # Scan matching
    odometry_edge_robust_kernel: "Huber" # NONE, Cauchy, DCS, Fair, GemanMcClure, Huber, PseudoHuber, Saturated, Tukey, Welsch
    odometry_edge_robust_kernel_size: 1.0
    loop_closure_edge_robust_kernel: "Huber"
    loop_closure_edge_robust_kernel_size: 1.0
    use_const_inf_matrix: false
    const_stddev_x: 0.5
    const_stddev_q: 0.1
    var_gain_a: 20.0
    min_stddev_x: 0.1
    max_stddev_x: 5.0
    min_stddev_q: 0.2
    max_stddev_q: 0.8

    graph_update_interval: 3.0
    keyframe_timer_update_interval: 500.0
    map_cloud_update_interval: 3.0
    map_cloud_resolution: 0.1
    main_timer_interval: 1.0
    
    infinite_room_dist_threshold: 1.0
    room_dist_threshold: 1.0

    extract_planar_surfaces:    true
    min_seg_points:             100
    use_euclidean_filter:       true
    use_shadow_filter:          false
    min_horizontal_inliers:     500
    min_vertical_inliers:       100
    keyframe_window_size:       5
    plane_information:          1.0 #0.1
    room_information:           1.0 #0.1
    corridor_information:       1.0 #0.1
    plane_dist_threshold:       0.35
    plane_points_dist:          0.5
    constant_covariance:        true
    min_plane_points:           100 # for plane extract
    dupl_plane_matching_information: 1.0 #0.1
    optimization_window_size: 5
    optimization_type: "GLOBAL"

    #output_file: "/root/workspaces/data/"
    save_timings: true
    save_info: true
    # loop detection
    use_loop_closure: true
    use_all_timer: true # loop detection for all timer or else only when duplicated room detected
    use_all_keyframe: true # loop detection for all keyframes from past or else only with kfs in duplicated room

    use_old_ablation: false